src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -104,6 +104,12 @@ ConcurrencyWorker::HandleExecuteOff()
       thread_config_->is_paused_ = true;
       std::unique_lock<std::mutex> lock(wake_mutex_);
       wake_signal_.wait(lock, [this]() { return early_exit || execute_; });
+
+      // TODO REFACTOR TMA-1043 - memory manager should be handling this instead
```

Comment:
Why would the memory manager do this? The number of active threads is used for correctly using the data loader, which is separate from the shared/non-shared memory type.

Reply:
Actually I can see how this might be separate from the data loader. I'm honestly a little confused on the boundaries between data loader and "memory manager"

Reply:
>I'm honestly a little confused on the boundaries between data loader and "memory manager"

That line isn't clear yet. The hope is that there is some single object created at the manager level that can hold this information instead of going from Manager->AllWorkers->AllContexts